aptos-move/framework/aptos-framework/sources/storage_gas.move

```diff
@@ -0,0 +1,254 @@
+module aptos_framework::storage_gas {
+
+    use aptos_framework::system_addresses;
+    use std::error;
+    use aptos_framework::state_storage;
+    use std::vector;
+
+    friend aptos_framework::gas_schedule;
+    friend aptos_framework::genesis;
+    friend aptos_framework::reconfiguration;
+
+    const ESTORAGE_GAS_CONFIG: u64 = 0;
+    const ESTORAGE_GAS: u64 = 1;
+    const EINVALID_GAS_RANGE: u64 = 2;
+    const EZERO_TARGET_UTILIZATION: u64 = 3;
+    const EINVALID_CURVE_LENGTH: u64 = 4;
+    const EINVALID_CURVE_START_POINT: u64 = 5;
+    const EINVALID_CURVE_END_POINT: u64 = 6;
+    const EINVALID_MONOTONICALLY_NON_DECREASING_CURVE: u64 = 7;
+
+    const BASIS_POINT_BASE: u64 = 10000;
+
+    /// This updates at reconfig and guarantees not to change elsewhere, safe
+    /// for gas calculation.
+    ///
+    /// Specifically, it is updated based on the usage at the begining of the
+    /// current epoch when end it by executing a reconfig transaction. The gas
+    /// schedule derived from these parameter will be for gas calculation of
+    /// the entire next epoch.
+    /// -- The data is one epoch stale than ideal, but VM doesn't need to worry
+    /// about reloading gas parameters after the first txn of an epoch.
+    struct StorageGas has key {
+        per_item_read: u64,
+        per_item_create: u64,
+        per_item_write: u64,
+        per_byte_read: u64,
+        per_byte_create: u64,
+        per_byte_write: u64,
+    }
+
+    // x and y are basis points.
+    struct Point has copy, drop, store {
+        x: u64,
+        y: u64
+    }
+
+    /// P(x) = min_price + (base ^ (utilization / target_utilization) - 1) / (base - 1) * (max_price - min_price)
+    // initialize the base to be 32, which means:
+    //   When DB is at 50% target utilization,the price increases roughly 15% of (max_price - min_price) on top of min_price.
+    //   More data points:
+    //     10% -> 1%
+    //     20% -> 3%
+    //     30% -> 6%
+    //     40% -> 10%
+    //     50% -> 15%
+    //     60% -> 23%
+    //     70% -> 33%
+    //     80% -> 48%
+    //     90% -> 70%
+    //     95% -> 84%
+    //     99% -> 96%
+    //    100% -> 100%
+    struct UsageGasConfig has copy, drop, store {
+        min_gas: u64,
+        max_gas: u64,
+        target_utilization: u64,
+        read_curve: vector<Point>,
+        create_curve: vector<Point>,
+        write_curve: vector<Point>,
+    }
+
+    struct StorageGasConfig has copy, drop, key {
+        item_config: UsageGasConfig,
+        byte_config: UsageGasConfig,
+    }
```

Comment:
I feel this should be target_items, target_bytes, plus 6 curves, and each curve is (min, max, points)